patroni/postgresql/mpp/__init__.py

```diff
@@ -65,6 +65,14 @@ def group(self) -> Any:
     def coordinator_group_id(self) -> Any:
         """The group id of the coordinator PostgreSQL cluster."""
 
+    @property
+    @abc.abstractmethod
+    def type(self) -> str:
+        """The type of the MPP cluster.
+
+        :returns: A string representation of the type of a given MPP implementation.
+        """
```

Comment:
I would suggest doing something like:
```suggestion
    @property
    def type(self) -> str:
        """The type of the MPP cluster.

        :returns: A string representation of the type of a given MPP implementation.
        """
        for base in self.__class__.__bases__:
            if not base.__name__.startswith('Abstract'):
                return base.__name__
        return self.__class__.__name__
```

This way we don't have to create additional properties in subclasses.